src/java.base/share/classes/java/util/Collections.java

```diff
@@ -1150,6 +1150,12 @@ public static <T> Set<T> unmodifiableSet(Set<? extends T> s) {
         if (s.getClass() == UnmodifiableSet.class) {
             return (Set<T>) s;
         }
```

Comment:
To avoid re‑wrapping already unmodifiable `EnumSet`s, this should also check for those:
```suggestion
        if (s.getClass() == UnmodifiableSet.class
            || s.getClass() == UnmodifiableRegularEnumSet.class
            || s.getClass() == UnmodifiableJumboEnumSet.class) {
            return (Set<T>) s;
        }
```